### PR TITLE
Use D3 Data lifecycle for Optimal DOM Manipulation; Closes #52

### LINKED ISSUE #52: Optimize DOM Manipulation
Currently the bodies directive completely removes all circles from the dom and adds new elements to represent the circles. We should use the d3 change lifecylce methods to simply update the dom values on each update.

### PR DESCRIPTION
**Pull request by benjic**
Problem
=======

Removing elements from the DOM and appending new ones once per animation frame
is suffering performance from performance issues.

Solution
========

Use the D3 lifecycle to update existing dom elements.

Howto Test
==========

- [ ] Run locally confirm no regressions

### CODE REVIEW

**Comment by mkinsey** [comment]
Looks good to me. Are you looking to merge this or continue work on it?

File: src/bridge/directives/bodies.js [hunk 1]
```diff
@@ -66,12 +112,19 @@ angular.module('bridge.directives')
             svg.select(".x.axis") .call(xAxis);
             svg.select(".y.axis").call(yAxis);
             bodyGroup.attr("transform", "translate(" + d3.event.translate + ") scale(" + d3.event.scale + ")");
-            zoneGroup.attr("transform", "translate(" + d3.event.translate + ") scale(" + d3.event.scale + ")");
+            zonesGroup.attr("transform", "translate(" + d3.event.translate + ") scale(" + d3.event.scale + ")");
           });
 
+
+        svg.call(zoom.translate([width/2, height/2]).event);
```
Comment (benjic) [inline src/bridge/directives/bodies.js]: These call to zoom events center the simulation in the middle of the viewport.

File: src/bridge/directives/bodies.js [hunk 2]
```diff
@@ -28,15 +75,14 @@ angular.module('bridge.directives')
...
-        var zoneGroup = svgGroup.append('g').attr('id', 'zoneGroup');
-
+        var zonesGroup = svgGroup.append('g').attr('id', 'zoneGroup');
         var bodyGroup = svgGroup.append('g').attr('id', 'bodyGroup');
 
-        // TODO: Get dimensions from element
-        var width  = document.getElementById('svg').offsetWidth;
-        var height = document.getElementById('svg').offsetHeight;
+        // Get bounding rect for the element
+        var rect = elem[0].firstChild.getBoundingClientRect();
```
Comment (benjic) [inline src/bridge/directives/bodies.js]: This removes the deprecated calls to `offset{Width, Height}`.

**Comment by benjic** [comment]
@mkinsey This is ready for review and merge. It may worthwhile to ask @alexander-dunn about integrating this into his effort.

**Comment by mkinsey** [comment]
Nice work @benjic! The performance is great, but I am finding the zooming erratic and hard to control.

**Comment by benjic** [comment]
I think it is due to translating but not centering the d3 zoom object. Good catch, let me fix.

**Comment by benjic** [comment]
@mkinsey this is ready for your inspection :)

**Pull request merged**